File: Catching Car Mileage Numbers.js
function isInteresting(number, awesomePhrases) {
  // Go to town!
  num = number.toString()
  
  //Function to check digit followed by all zeors
  function allZero(num) {
    var a = []
    const checkZero = x => x == 0
    
    for(let i = 1; i < num.length; i++) {
      a.push(num[i])
    }
    
    return (num[0] > 0 && a.every(checkZero))
  }
  
  //Function to check every digit is the same number
  function allSame(num) {
    var a = []
    for(let i = 0; i < num.length; i++) {
      a.push(num[i])
    }
    
    const checkSame = x => x == a[0]
    
    return a.every(checkSame)
  }
  
  //Function to check for sequential
  function sequential(num) {
    const alpha = "1234567890"
    const beta = "9876543210"
    
    return (alpha.includes(num) || beta.includes(num))
  }
  
  //Function to check if the number is palindrome
  function palindrome(num) {
    var a = num.split("").reverse()
    a = a.join("")
    return (a == num)
  }
  
  //Function to check if the number is found in the array
  function match(num, arr) {
    num = parseInt(num)
    return arr.includes(num)
  }
  
  //Function to check if the number is interesting
  function checkInteresting(num, arr) {
    if(num.length >= 3) {
      if(allZero(num) || allSame(num) || sequential(num) || palindrome(num) || match(num, arr)){
        return true
      } else {
        return false
      }
    }
    return false
  }

  
  function finalCheck(num, arr) {
    if(checkInteresting(num, arr)) {
      return 2
    } else if(checkInteresting((parseInt(num) + 1).toString(), arr) || checkInteresting((parseInt(num) + 2).toString(), arr)) {
      return 1
    } else {
      return 0
    }
  }
  
  return finalCheck(num, awesomePhrases)
}

File: Catching Car Mileage Numbers.py
def is_interesting(number, awesome_phrases):
    num = str(number)
    
    def allZero(num):
        check = True
        a = list(num[i] for i in range(1, len(num)))
        for i in a:
            if(i != "0"):
                check = False
                break
        
        return check
    
    def allSame(num):
        check = True
        a = list(num)
        for i in a:
            if(i != a[0]):
                check = False
                break
        
        return check
    
    def sequential(num):
        alpha = "1234567890"
        beta = "9876543210"
        return (num in alpha or num in beta)
    
    def palindrome(num):
        return num[0:] == num[::-1]
    
    def match(num, arr):
        return int(num) in arr
    
    def checkInteresting(num, arr):
        if(len(num) >= 3):
            if(allZero(num) or allSame(num) or sequential(num) or palindrome(num) or match(num, arr)):
                return True
            else:
                return False
        else:
            return False
    
    def finalCheck(num, arr):
        if(checkInteresting(num, arr)):
            return 2
        elif (checkInteresting(str(int(num) + 1), arr) or checkInteresting(str(int(num) + 2), arr)):
            return 1
        else:
            return 0
    
    return finalCheck(num, awesome_phrases)
            
